net/packet/packet.go

```diff
@@ -502,3 +536,53 @@ func withIP(ap netip.AddrPort, ip netip.Addr) netip.AddrPort {
 func withPort(ap netip.AddrPort, port uint16) netip.AddrPort {
 	return netip.AddrPortFrom(ap.Addr(), port)
 }
+
+// updateV4PacketChecksums updates the checksums in the packet buffer.
+// Currently (2023-03-01) only TCP/UDP/ICMP over IPv4 is supported.
+// The packet is modified in place.
+func updateV4PacketChecksums(p *Parsed, old, new netip.Addr) {
```

Comment:
Add to docs that the `*Parsed` must have been decoded as an IPProto != Unknown or this is unsafe.